interfaces/builtin/mir.go

```diff
@@ -0,0 +1,606 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (c) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more dtails.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.
+ *
+ */
+
+package builtin
+
+import (
+       "github.com/snapcore/snapd/interfaces"
+)
+
+var mirPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the Mir server. Reserved because this
+# gives privileged access to the system.
+# Usage: reserved
+
+
+capability dac_override,
```

Comment:
without it the following occurs on server launch

Jun 16 21:19:13 localhost ./devtools.snapd[5034]: taskrunner.go:238: DEBUG: Running task 100 on Do: Make snap "mir-server" available to the system
Jun 16 21:19:13 localhost systemd[1]: Reloading.
Jun 16 21:19:14 localhost systemd[1]: snapd.refresh.timer: Adding 1h 9min 46.277746s random time.
Jun 16 21:19:14 localhost systemd[1]: Reloading.
Jun 16 21:19:14 localhost systemd[1]: snapd.refresh.timer: Adding 9min 58.505083s random time.
Jun 16 21:19:14 localhost systemd[1]: Started Service for snap application mir-server.mir-server.
Jun 16 21:19:14 localhost ./devtools.snapd[5034]: daemon.go:181: DEBUG: uid=0;@ GET /v2/snaps 1.126997ms 200
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: Mir server snap started
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.438531] mirplatform: Found graphics driver: mir:mesa-kms (version 0.23.0)
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.455655] mirplatform: Found graphics driver: mir:mesa-x11 (version 0.23.0)
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.460754] mirserver: Starting
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.461495] mircommon: Loading modules from: /snap/mir-server/x1/usr/lib/x86_64-linux-gnu/mir/server-platform
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.461990] mircommon: Loading module: /snap/mir-server/x1/usr/lib/x86_64-linux-gnu/mir/server-platform/graphics-mesa-kms.so.9
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.462408] mircommon: Loading module: /snap/mir-server/x1/usr/lib/x86_64-linux-gnu/mir/server-platform/server-mesa-x11.so.9
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.463333] mircommon: Loading module: /snap/mir-server/x1/usr/lib/x86_64-linux-gnu/mir/server-platform/input-evdev.so.5
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.465184] mirplatform: Found graphics driver: mir:mesa-kms (version 0.23.0)
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.465662] mirplatform: Found graphics driver: mir:mesa-x11 (version 0.23.0)
Jun 16 21:19:14 localhost ubuntu-core-launcher[5138]: [2016-06-16 21:19:14.466394] mirserver: Selected driver: mir:mesa-kms (version 0.23.0)
Jun 16 21:19:14 localhost kernel: [ 4508.651864] audit: type=1400 audit(1466111954.459:24): apparmor="DENIED" operation="capable" profile="snap.mir-server.mir-server" pid=5140 comm="mir-server" capability=1  capname="dac_override"
Jun 16 21:19:14 localhost kernel: [ 4508.651871] audit: type=1400 audit(1466111954.459:25): apparmor="DENIED" operation="capable" profile="snap.mir-server.mir-server" pid=5140 comm="mir-server" capability=2  capname="dac_read_search"